util/EIP-712/src/encode.rs

```diff
@@ -56,11 +56,17 @@ fn build_dependencies<'a>(message_type: &'a str, message_types: &'a MessageTypes
 			deps.insert(item);
 
 			for field in fields {
+				// check if this field is an array type
+				let field_type = if let Some (index) = field.type_.find('[') {
```

Comment:
```suggestion
				let field_type = if let Some(index) = field.type_.find('[') {
```
funny space after `Some`